--- ./gtk/inspector/general.c.orig	2022-08-24 15:26:58.000000000 -0700
+++ ./gtk/inspector/general.c	2022-08-24 15:29:41.000000000 -0700
@@ -40,7 +40,6 @@
 #ifdef GDK_WINDOWING_X11
 #include "x11/gdkx.h"
 #include <epoxy/glx.h>
-#include <epoxy/egl.h>
 #endif
 
 #ifdef GDK_WINDOWING_WIN32
@@ -289,37 +288,6 @@
 }
 #endif
 
-#if defined(GDK_WINDOWING_WAYLAND) || defined (GDK_WINDOWING_X11) || (defined (GDK_WINDOWING_WIN32) && defined(GDK_WIN32_ENABLE_EGL))
-static void
-append_egl_extension_row (GtkInspectorGeneral *gen,
-                          EGLDisplay          dpy,
-                          const char          *ext)
-{
-  add_check_row (gen, GTK_LIST_BOX (gen->gl_box), ext, epoxy_has_egl_extension (dpy, ext), 0);
-}
-
-static EGLDisplay
-get_egl_display (GdkDisplay *display)
-{
-#ifdef GDK_WINDOWING_WAYLAND
-  if (GDK_IS_WAYLAND_DISPLAY (display))
-    return gdk_wayland_display_get_egl_display (display);
-  else
-#endif
-#ifdef GDK_WINDOWING_X11
-  if (GDK_IS_X11_DISPLAY (display))
-    return gdk_x11_display_get_egl_display (display);
-  else
-#endif
-#ifdef GDK_WINDOWING_WIN32
-  if (GDK_IS_WIN32_DISPLAY (display))
-    return gdk_win32_display_get_egl_display (display);
-  else
-#endif
-   return NULL;
-}
-#endif
-
 static void
 init_gl (GtkInspectorGeneral *gen)
 {
@@ -341,25 +309,6 @@
       return;
     }
 
-#if defined(GDK_WINDOWING_X11) || defined(GDK_WINDOWING_WAYLAND) || (defined(GDK_WINDOWING_WIN32) && defined(GDK_WIN32_ENABLE_EGL))
-  EGLDisplay egl_display = get_egl_display (gen->display);
-  if (egl_display)
-    {
-      char *version;
-
-      version = g_strconcat ("EGL ", eglQueryString (egl_display, EGL_VERSION), NULL);
-      gtk_label_set_text (GTK_LABEL (gen->gl_version), version);
-      g_free (version);
-
-      gtk_label_set_text (GTK_LABEL (gen->gl_vendor), eglQueryString (egl_display, EGL_VENDOR));
-
-      append_egl_extension_row (gen, egl_display, "EGL_KHR_create_context");
-      append_egl_extension_row (gen, egl_display, "EGL_EXT_buffer_age");
-      append_egl_extension_row (gen, egl_display, "EGL_EXT_swap_buffers_with_damage");
-      append_egl_extension_row (gen, egl_display, "EGL_KHR_surfaceless_context");
-    }
-  else
-#endif
 #ifdef GDK_WINDOWING_X11
   if (GDK_IS_X11_DISPLAY (gen->display))
     {
