diff --git Source/WTF/CMakeLists.txt Source/WTF/CMakeLists.txt
index 7c615c55..dd221b1a 100644
--- Source/WTF/CMakeLists.txt
+++ Source/WTF/CMakeLists.txt
@@ -6,6 +6,6 @@ add_subdirectory(wtf)
 
 # Apple builds have the ICU headers checked into ${WTF_DIR}/icu
 # Copy them into ${ICU_INCLUDE_DIRS} so the build behaves like find_package was used
-if (APPLE)
+if (DISABLEAPPLE)
     file(COPY ${WTF_DIR}/icu/unicode DESTINATION ${ICU_INCLUDE_DIRS})
 endif ()
