--- Makefile.in.orig	2022-09-17 11:45:11
+++ Makefile.in	2022-10-29 11:49:03
@@ -10,7 +10,7 @@
 GCOPTIONS = @GCOPTIONS@
 GCLIB = @GCLIB@
 GCPPLIB = @GCPPLIB@
-GCLIBS = $(GCPPLIB) $(GCLIB)
+GCLIBS = $(GCPPLIB) $(GCLIB) -lgccpp
 LFLAGS = @LDFLAGS@ -LLspCpp -LLspCpp/third_party/uri/src
 LIBS = $(LFLAGS) @PTHREAD_LIBS@ @GLEW@ @LIBS@ $(GCLIBS) @LSPLIBS@
 DOSLIBS = $(subst -lncurses, -ltermcap, $(LIBS)) -lwolfssl -lgdi32 -lwinmm -s -static
@@ -149,7 +149,7 @@
 	echo @set Datadir @datadir@ >> doc/version.texi
 	echo @set Docdir @docdir@ >> doc/version.texi
 
-	echo "#!/usr/bin/env python3" > GUI/xasyVersion.py
+	echo "#!@@PYTHON@@" > GUI/xasyVersion.py
 	echo xasyVersion = \"$(revision)\" >> GUI/xasyVersion.py
 
 	if test -n "$(MSDOS)"; then \
